test-result-summary-client/src/PerfCompare/PerfCompare.jsx

```diff
@@ -429,34 +464,54 @@ export default class PerfCompare extends Component {
                     continue;
                 }
 
+                // get the raw values for master nodes.
                 curMetricName = this.state.benchmarkRuns.benchmarkRunBaseline.parsedVariants[i].metrics[j].name;
+                let childBuildBaselineData = {};
+                let childBuildTestData = {};
+                if ( this.state.benchmarkRuns.childBuildsListBaseline && this.state.benchmarkRuns.childBuildsListBaseline.length > 0){
+                    for ( let x = 0; x < this.state.benchmarkRuns.childBuildsListBaseline.length; x++ ) {
+                        childBuildBaselineData["ite " + (x + 1) ] = this.state.benchmarkRuns.childBuildsListBaseline[x][j].value;
+                    }
+                }
+                if (this.state.benchmarkRuns.childBuildsListTest && this.state.benchmarkRuns.childBuildsListTest.length > 0){
+                    for ( let y = 0; y < this.state.benchmarkRuns.childBuildsListTest.length; y++ ) {
+                        childBuildTestData["ite " + (y + 1) ] = this.state.benchmarkRuns.childBuildsListTest[y][j].value;
+                    }
+                }
 
+                // raw values displaying: master jobs diplay all their children raw values, and child jobs diplay their own raw values.
                 curRawValues = {
-                    "baseline": this.state.benchmarkRuns.benchmarkRunBaseline.parsedVariants[i].metrics[j].value,
-                    "test": this.state.benchmarkRuns.benchmarkRunTest.parsedVariants[curMatchingTestVariantIndex].metrics[curMatchingTestMetricIndex].value
+                    "baseline": {
+                        "overview": this.state.benchmarkRuns.benchmarkRunBaseline.parsedVariants[i].metrics[j].value,
+                        "iteData": (this.state.benchmarkRuns.benchmarkRunBaseline.parsedVariants[i].testsData === undefined) ? childBuildBaselineData : this.state.benchmarkRuns.benchmarkRunBaseline.parsedVariants[i].testsData[j].value
+                     },
+                    "test": {
+                        "overview": this.state.benchmarkRuns.benchmarkRunTest.parsedVariants[curMatchingTestVariantIndex].metrics[curMatchingTestMetricIndex].value,
+                        "iteData": (this.state.benchmarkRuns.benchmarkRunTest.parsedVariants[i].testsData === undefined) ? childBuildTestData : this.state.benchmarkRuns.benchmarkRunTest.parsedVariants[i].testsData[j].value
```

Comment:
We need to take care of the scenario when we have multiple tests running in one Jenkins jobs. Hence, we'll need to show all the tests from that Jenkins run.